src/components/utils/GetUserByld.ts

```diff
@@ -0,0 +1,19 @@
+import { User } from '../types/User';
+import usersFromServer from '../../api/users';
+
+const anonymousUser: User = {
+  id: 0,
+  name: 'Anonymous',
+  username: 'anon',
+  email: 'anonymous@unknown.com',
+};
+
+export function getUserById(id: number): User | undefined {
+  const foundUser = usersFromServer.find(u => u.id === id);
+  /* eslint-disable no-console */
+
+  console.warn(`User with ID ${id} not found. Returning anonymous user.`);
```

Comment:
Consider moving the console warning inside a conditional block to only execute when the user is not found. This will prevent unnecessary warnings when a user is successfully found.